### PR TITLE
[karmadactl get] judge with label on the resources determine whether resource is adopted by Karmada

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

We have added labels to the resources in the member cluster to indicate whether the current resource is propagated by Karmada:

https://github.com/karmada-io/karmada/blob/1953a0ee50d9d452b9ca5fffdb2f60338eed0b63/pkg/util/constants.go#L26-L30

Therefore, we can directly process this label in the `karmadactl get` command, without needing to retrieve RB/CRB resources.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

One reason for this change is that we cannot assume that users using karmadactl by default have the authority to get rb/crb resources.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by lonelyCZ** [comment]
Looks nice, thanks for your work, I will look it ASAP.

/assign

**Comment by RainbowMango** [comment]
> One reason for this change is that we cannot assume that users using karmadactl by default have the authority to get rb/crb resources.

+1

File: pkg/karmadactl/get/get.go [hunk 1]
```diff
@@ -578,20 +571,25 @@ func (g *CommandGetOptions) reconstructObj(obj runtime.Object, mapping *meta.RES
...
+		unObj := &unstructured.Unstructured{}
+		err := unObj.UnmarshalJSON(table.Rows[rowIdx].Object.Raw)
+		if err != nil {
+			klog.Errorf("Failed to unmarshal unObj, error is: %v", err)
+			continue
+		}
+		v, exist := unObj.GetLabels()[karmadautil.ManagedByKarmadaLabel]
+		if exist && v == karmadautil.ManagedByKarmadaLabelValue {
+			table.Rows[rowIdx].Cells = append(table.Rows[rowIdx].Cells, "Y")
 		}
```
Comment (zishen) [inline pkg/karmadactl/get/get.go:591]: Here need a `continue`?  or `Y` and `N` can appear at the same time?

Comment (XiShanYongYe-Chang) [reply pkg/karmadactl/get/get.go:593]: Thanks for reminding.

File: pkg/karmadactl/get/get.go [hunk 2]
```diff
@@ -548,16 +533,24 @@ func (g *CommandGetOptions) reconstructionRow(objs []Obj, table *metav1.Table) (
...
-			if _, ok := RBInfo[rbKey]; ok {
-				tempRow.Cells = append(tempRow.Cells, "Y")
-			} else {
-				tempRow.Cells = append(tempRow.Cells, "N")
+			v, exist := unObj.GetLabels()[karmadautil.ManagedByKarmadaLabel]
+			if exist && v == karmadautil.ManagedByKarmadaLabelValue {
+				table.Rows[rowIdx].Cells = append(table.Rows[rowIdx].Cells, "Y")
 			}
-			table.Rows[rowIdx].Cells = tempRow.Cells
+			table.Rows[rowIdx].Cells = append(table.Rows[rowIdx].Cells, "N")
```
Comment (lonelyCZ) [inline pkg/karmadactl/get/get.go:553]: It also need `else` branch.

**Review by lonelyCZ — COMMENTED** [review]
I just tested it in my env, it worked fine.

File: pkg/karmadactl/get/get.go (hunk 2 above)
Comment (XiShanYongYe-Chang) [reply pkg/karmadactl/get/get.go]: Thanks~, updated.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**